Sprint 12

DmitriyBrunkov opened the pull request.

elayhan committed:
feat:добавить функциональность - общие фильмы (#10)

* feat:добавить функциональность - общие фильмы

* style: fix

* style: fix2

NullDave committed:
feat: добавить режиссёров в фильмы

* feat: Добавление режиссёров в фильмы

* style: ExceptionController fix

* style: DirectorService fix

* style: pain

* refactor: исправлены замечания

elayhan committed:
feat: удаление фильмов и пользователей

* feat: удаление фильмов и пользователей

* style: fix

DmitriyBrunkov committed:
feat: добавить функциональность Рекомендации

* feat: добавление функциональности Рекомендации

* rebase fix

* codestyle fix

elayhan committed:
feat: реализовать вывод самых популярных фильмов по жанру и годам

* feat: реализовать вывод самых популярных фильмов по жанру и годам

* style: fix

destinyApproaching committed:
feat: добавить функциональность Поиск

ciiska5 committed:
feat: добавить функциональность - отзывы

* feat: добавить функциональность - отзывы

* fix: поправить синтаксис

* fix: поправить синтаксис 2

* fix: поправить синтаксис 3 :)

* fix: поправить синтаксис в filmService

* fix: поправить синтаксис в ReviewDbStorage

* fix: поправить порядок дропа таблиц

* fix: удалить повтор дропа таблицы films_diresctors

* style: поменять название Reviews на Review

---------

Co-authored-by: e1ayhan <wwwelayhan@yandex.ru>

destinyApproaching committed:
feat: Функциональность «Лента событий»

* feat: Функциональность «Лента событий»

* Удалил db из своей репы

* style: УДАЛЁН ЛИШНИЙ ИМПОРТ

* style: УДАЛЁН ЕЩЁ ОДИН ЛИШНИЙ ИМПОРТ

* style: ДОБАВЛЕН НОВАЯ ПУСТАЯ СТРОКА

* style: УДАЛЁН ПРОБЕЛ

* style: удалён комментарий, к сожаления, ЧЕКСТАЙЛА НА НИХ НЕТ

DmitriyBrunkov committed:
Refactor

* refactor: db scheme

* refactor: move FilmStorage from UserService to UserStorage

* refactor: Queries typo

* refactor: readme scheme update

* refactor: handle warnings

* refactor: add String constraints

* refactor: misc

DmitriyBrunkov committed:
refactor: checkstyle fix

DmitriyBrunkov committed:
refactor: double like fix

Kapinosva commented on src/main/java/ru/yandex/practicum/filmorate/storage/feed/FeedQueries/FeedQueries.java:
```diff
@@ -0,0 +1,8 @@
+package ru.yandex.practicum.filmorate.storage.feed.FeedQueries;
+
+public final class FeedQueries {
```
классам хелперам, от которых не требуется создание объекта, принято делать приватный констурктор,

Kapinosva reviewed (commented):
Очень хорошая раота, вы большие молодцы. Добавьте плиз описание тго, что сделали, в текст коммита или в ридми.

И всем классам помощникам надо делать приватный конструктор

DmitriyBrunkov committed:
refactor: add private constructors to helper classes and add project description:

Команда разработчиков №12 добавила в него новый, восхитительный функционал:
1. Функциональность «Общие фильмы»
   - вывод общих с другом фильмов с сортировкой по их популярности
2. Функциональность «Лента событий»
    - возможность просмотра последних событий на платформе
3. Добавление режиссёров в фильмы
    - добавлена информация о режиссерах фильма и вывод фильмов по режиссерам с сортиовками
4. Функциональность «Рекомендации»
    - реализована простая рекомендательная система для фильмов
5. Удаление фильмов и пользователей
    - добавлена функциональность для удаления фильма и пользователя по идентификатору
6. Вывод самых популярных фильмов по жанру и годам
    - добавлена возможность выводить топ фильмов по количеству лайков с фильтрацией
7. Функциональность «Поиск»
    - реализован поиск по названию фильмов и по режиссёру
8. Функциональность «Отзывы»
    - добавлены отзывы на фильмы с рейтингом

Kapinosva reviewed (approved):
Все супер. большие молодцы. Хорошая работа. Успехов вам всем в дальнейшей работе программитсами!!
